src/com/owncloud/android/authentication/AuthenticatorActivity.java

```diff
@@ -305,7 +333,56 @@ public void onClick(View v) {
         /// initialize block to be moved to single Fragment to retrieve and validate credentials 
         initAuthorizationPreFragment(savedInstanceState);
 
-        //Log_OC.e(TAG,  "onCreate end");
+        mCustomTabPackageName = getCustomTabPackageName();
+
+        if(mCustomTabPackageName != null) {
+            CustomTabsClient.bindCustomTabsService(this, mCustomTabPackageName, mCustomTabServiceConnection);
+        }
+    }
+
+    @Override
+    public void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        // check if its a oauth SALS callback intent
+        if(intent.getAction().equals(ACTION_VIEW) &&
+                intent.getData().getHost().equals("android.owncloud.com")) {

```

Comment:
Isn't there a different way to get "android.owncloud.com" instead of using a hardcoded string?

Reply:
well its the redirect for oauth.
https://github.com/owncloud/android/blob/1c40bad0dd70a76046fabec3c4a9ae2e87477f9c/AndroidManifest.xml#L203

As long as i know it was hard coded before. We could put it into a constant if thats better.